### PR TITLE
Login(feat): Create email account 

### PR DESCRIPTION
**Pull request by meunomeecris**
This PR contains the follow PR:
https://github.com/viniciusaro/evidence/pull/42

## Description, motivation, and context

- Add flow to reset password 
- Show confirmation with popUp

## Screenshots and GIFs (Artifacts)
Simulator Screenshot - iPhone 15 - 2024-05-13 at 10 32 23
Simulator Screenshot - iPhone 15 - 2024-05-13 at 10 32 13
Simulator Screenshot - iPhone 15 - 2024-05-13 at 10 32 17

## Checklist :leaves:

- [x ] Created all unit tests needed
- It's missing the unit test for Firebase

Thank you!

### CODE REVIEW

File: mobile/evidence/ios/Features/Sources/Login/LoginResetPasswordViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        } else if isResetPasswordButtonPressed == true && (isValidEmail == false) {
+            return "Email not valid."
+        }
+        return nil
+    }
+
+    func isValidEmail(_ email: String) -> Bool {
+        let regex = try! NSRegularExpression(pattern: "^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,}$", options: [.caseInsensitive])
+        return regex.firstMatch(in: email, options: [], range: NSRange(location: 0, length: email.utf16.count)) != nil
+    }
```
Comment (viniciusaro) [inline mobile/evidence/ios/Features/Sources/Login/LoginResetPasswordViewModel.swift:77]: We are repeating this logic in many places. Maybe we could create a new dependency called "InputValidator" that can extract this kind of logic out of our view models, what do you think?

File: mobile/evidence/ios/Features/Tests/LoginTests/LoginEmailViewModelTests.swift [hunk 2]
```diff
@@ -11,101 +11,102 @@ import XCTest
 
 final class LoginEmailViewModelTests: XCTestCase {
 
+    func testConfirmationPopupAppears() {
+        let viewModel = LoginEmailViewModel()
+        viewModel.confirmationPopupAppears()
+        let expectation = XCTestExpectation(description: "Confirmation popup should appear")
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) {
+            XCTAssertEqual(viewModel.offSetY, 1000, "OffsetY should be updated to 1000")
```
Comment (viniciusaro) [inline mobile/evidence/ios/Features/Tests/LoginTests/LoginEmailViewModelTests.swift:20]: Let's talk about this delay in tests! The swift-dependencies package ships with a [clock](https://github.com/pointfreeco/swift-dependencies?tab=readme-ov-file#quick-start) dependency that can be used to allow waiting some time to pass during user experience, but allowing it to happen instantly in unit tests.

File: mobile/evidence/ios/Features/Tests/LoginTests/LoginEmailViewModelTests.swift [hunk 3]
```diff
@@ -11,101 +11,102 @@ import XCTest
...
+        viewModel.emailInput = "email@valid.com"
+        viewModel.isLoginEmailButtonPressed = true
+
         viewModel.clearEmailInputTapped()
-<<<<<<< HEAD
-        XCTAssertTrue(viewModel.loginEmailInput.isEmpty, "Should be empty")
-=======
         XCTAssertTrue(viewModel.emailInput.isEmpty, "Should be empty")
->>>>>>> main
-        XCTAssertFalse(viewModel.isNextButtonPressed,  "Should set false")
```
Comment (viniciusaro) [inline mobile/evidence/ios/Features/Tests/LoginTests/LoginEmailViewModelTests.swift:32]: Great!

File: mobile/evidence/ios/Leaf/Sources/Leaf/Components/LeafButton.swift [hunk 4]
```diff
@@ -72,18 +73,40 @@ private struct LeafGoogleLoginButton: View {
...
+            Button("Reset password") {
 
             }
-            .buttonStyle(LeafGoogleLoginButtonStyle())
+            .buttonStyle(LeaflinkButtonStyle())
 
         }
         .padding(20)
-        .background(.green)
+//        .background(.green)
```
Comment (viniciusaro) [inline mobile/evidence/ios/Leaf/Sources/Leaf/Components/LeafButton.swift:109]: Pode apagar ao invés de comentar

**Review by viniciusaro — APPROVED** [review]
Great, my friend!!! 👏👏👏👏

File: mobile/evidence/ios/Features/Sources/Login/LoginEmailViewModel.swift [hunk 5]
```diff
@@ -17,6 +17,8 @@ final public class LoginEmailViewModel: ObservableObject, Identifiable {
     @Published private(set) var isValidPassword: Bool
     @Published var isLoginEmailButtonPressed: Bool
     @Published var isEmailInputFocused: Bool
+    @Published var loginResetPassword: LoginResetPasswordViewModel?
+    @Published private(set) var offSetY: CGFloat
```
[the remark this answers is not in the document]
Comment (meunomeecris) [reply mobile/evidence/ios/Features/Sources/Login/LoginEmailViewModel.swift:21]: Definitely! Today I learned the difference between "alert and error". Cool.

File: mobile/evidence/ios/Features/Sources/Login/LoginResetPasswordViewModel.swift (hunk 1 above)
Comment (meunomeecris) [reply mobile/evidence/ios/Features/Sources/Login/LoginResetPasswordViewModel.swift:77]: Discord >

File: mobile/evidence/ios/Features/Tests/LoginTests/LoginEmailViewModelTests.swift (hunk 2 above)
Comment (meunomeecris) [reply mobile/evidence/ios/Features/Tests/LoginTests/LoginEmailViewModelTests.swift:20]: yes chef!

File: mobile/evidence/ios/Features/Tests/LoginTests/LoginEmailViewModelTests.swift (hunk 3 above)
Comment (meunomeecris) [reply mobile/evidence/ios/Features/Tests/LoginTests/LoginEmailViewModelTests.swift:32]: When I merged the PR43 those conflicts disappeared. 🗡️

**Pull request merged**